src/funjson.c

```diff
@@ -504,37 +519,27 @@ json_map_call(ufun_attrib *ufun, sqlite3_str *rbuff, PE_REGS *pe_regs,
     pe_regs_setenv(pe_regs, 1, (const char *) sqlite3_column_text(json, 1));
   } else {
     const char *utf8;
-    char *latin1, *c;
+    char *latin1;
     int ulen, len;
 
     utf8 = (const char *) sqlite3_column_text(json, 1);
     ulen = sqlite3_column_bytes(json, 1);
     latin1 = utf8_to_latin1_us(utf8, ulen, &len, 0, "string");
-    for (c = latin1; *c; c += 1) {
-      if (!isprint(*c) && !isspace(*c)) {
-        *c = '?';
-      }
-    }
-
-    pe_regs_setenv(pe_regs, 1, latin1);
-    mush_free(latin1, "string");
+    char * partialbuf = json_unescape_latin1string(latin1,len);
+    mush_free(partialbuf, "json.string.partialbuf");
+    pe_regs_setenv(pe_regs, 1, partialbuf);
+    mush_free(latin1, "json.string");
   }
   if (sqlite3_column_type(json, 2) == SQLITE_INTEGER) {
     pe_regs_setenv(pe_regs, 2, pe_regs_intname(sqlite3_column_int(json, 2)));
   } else if (sqlite3_column_type(json, 2) == SQLITE_TEXT) {
     const char *utf8 = (const char *) sqlite3_column_text(json, 2);
     int ulen = sqlite3_column_bytes(json, 2);
     int len;
-    char *c;
     char *latin1 = utf8_to_latin1(utf8, ulen, &len, 0, "string");
-
-    for (c = latin1; *c; c += 1) {
-      if (!isprint(*c) && !isspace(*c)) {
-        *c = '?';
-      }
-    }
-
-    pe_regs_setenv(pe_regs, 2, latin1);
+    char * partialbuf = json_unescape_latin1string(latin1,len);
+    pe_regs_setenv(pe_regs, 2, partialbuf);
+    mush_free(partialbuf, "json.string.partialbuf");
```

Comment:
Using partialbuf immediately after freeing it

Reply:
Good catch